signing_service/src/tests/tests.py

```diff
@@ -0,0 +1,59 @@
+from unittest import TestCase
+import mock
+import sys
+
+from ..constants import SIGNING_SERVICE_DEFAULT_PORT
+from ..signing_service import _parse_arguments
+from ..signing_service import main
+
+
+class SigningServiceMainTestCase(TestCase):
+
+    def test_that_main_should_called_with_correct_address(self):
+        with self.assertRaises(SystemExit):
+            main('not_really_ip_or_host', 8000)
+
+    def test_that_main_should_exit_when_called_with_wrong_address(self):
+        with mock.patch('socket.socket.connect') as mock_socket_connect:
+            with mock.patch('socket.socket.send') as mock_socket_send:
+                with mock.patch('socket.socket.recv') as mock_socket_recv:
+                    with mock.patch('socket.socket.close') as mock_socket_close:
+                        main('127.0.0.1', 8000)
+
+        mock_socket_connect.assert_called_once_with(('127.0.0.1', 8000))
+        mock_socket_send.assert_called_once()
+        mock_socket_recv.assert_called_once()
+        mock_socket_close.assert_called_once()
+
+
+class SigningServiceParseArgumentsTestCase(TestCase):
+
+    def setUp(self):
+        super().setUp()
+        sys.argv = sys.argv[:1]
```

Comment:
I am not sure if understand what is it for ;)

Reply:
`ArgumentParser` takes values directly from `sys.argv`, but the test runner has its own arguments, so they have to be replaced.

Reply:
This should be commented in the code.

Reply:
:+1: done